src/oxygen/robot3_interface.py

```diff
@@ -0,0 +1,378 @@
+        return local_zone.utcoffset(None)
+
+
+    def get_keywords_status(self, *keywords):
+        '''
+        keywords: List of Robot keywords
+
+        Return: 'PASS' or 'FAIL'
+        '''
+        if sum(not kw.passed for kw in filter(None, keywords)):
+            return 'FAIL'
+        else:
+            return 'PASS'
+
+
+    def create_wrapper_keyword(self,
+                               name,
+                               start_timestamp,
+                               end_timestamp,
+                               setup,
+                               *keywords):
+        status = self.get_keywords_status(*keywords)
+        start_time = self.timestamp_to_ms(start_timestamp)
+        end_time = self.timestamp_to_ms(end_timestamp)
+
+        robot_keyword = self.spawn_robot_keyword(name,
+                                                 [],
+                                                 status,
+                                                 start_time,
+                                                 end_time,
+                                                 None,
+                                                 keywords,
+                                                 [],
+                                                 setup,
+                                                 (not setup))
+
+        return robot_keyword
+
+
+class RobotRunningInterface(object):
+    def build_suite(self, parsed_results):
+        robot_root_suite = RobotRunningSuite(parsed_results['name'])
+        for parsed_suite in parsed_results['suites']:
+            robot_suite = robot_root_suite.suites.create(parsed_suite['name'])
+            for subsuite in parsed_suite.get('suites', []):
+                robot_subsuite = self.build_suite(subsuite)
+                robot_suite.suites.append(robot_subsuite)
+            self.build_tests(parsed_suite, robot_suite)
+        self.build_tests(parsed_results, robot_root_suite)
+        return robot_root_suite
+
+    def build_tests(self, oxygen_suite, robot_suite):
+        for parsed_test in oxygen_suite.get('tests', []):
+            name = parsed_test['name']
+            tags = parsed_test['tags']
+            kw = parsed_test['keywords'][0]
+            msg = '\n'.join(kw['messages'])
+            test_robot_counterpart = robot_suite.tests.create(name, tags=tags)
+            if kw['pass']:
+                args = [msg if msg else 'Test passed :D']
+                test_robot_counterpart.keywords.create('Pass execution',
+                                                       args=args)
+            else:
+                args = [msg if msg else 'Test failed D:']
+                test_robot_counterpart.keywords.create('Fail', args=args)
+
+
+class RobotKeywordInterface(object):
+    KEYWORD_TYPE = RobotResultKeyword.KEYWORD_TYPE
+    SETUP_TYPE = RobotResultKeyword.SETUP_TYPE
+    TEARDOWN_TYPE = RobotResultKeyword.TEARDOWN_TYPE
```

Comment:
Why are we exposing this new functionality? What outside of interface needs this information?